sympy/printing/latex.py

```diff
@@ -1566,6 +1566,15 @@ def parens(x):
             return self._print(x)
         return r' \circ '.join(map(parens, expr.args))
 
+    def _print_HadamardPower(self, expr):
+        from sympy import Add, MatAdd, MatMul
+
+        def parens(x):
+            if isinstance(x, (Add, MatAdd, MatMul)):
```

Comment:
Can `parenthesize()`, `PRECEDENCE`, etc work here instead of hardcoding a list of classes?

(Also in `_print_HadamardProduct`)

Reply:
Yes, take example from the other printers.

Reply:
Another idea is to create a new method, for example `_power_print_helper`, and have both `_print_Pow`, `_print_HadamardPower`, and `_print_MatPow` call that method.

This would avoid repeating many times the code to print power-expressions.